core/src/main/java/nu/yona/server/messaging/service/MessageService.java

```diff
@@ -52,7 +51,7 @@
 	private BuddyService buddyService;
 
 	@Autowired
-	private UserAnonymizedService userAnonymizedService;
+	private BatchProxyService batchProxyService;
```

Comment:
Let's call this from the admin service instead of the core, as this functionality is really admin specific.